zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java

```diff
@@ -725,8 +726,31 @@ private static JobResult getJobResultOrDefault(final JobCompletionRequest reques
     if (request == null || request.getResult() == null) {
       return new JobResult();
     }
-    return new JobResult()
-        .setDenied(getBooleanOrDefault(request, r -> r.getResult().getDenied(), false));
+
+    final JobResult jobResult = new JobResult();
+    jobResult.setDenied(getBooleanOrDefault(request, r -> r.getResult().getDenied(), false));
+
+    jobResult.setCorrectedAttributes(
+        getStringListOrEmpty(request, r -> r.getResult().getCorrectedAttributes()));
+
+    if (request.getResult().getCorrections() == null) {
+      return jobResult;
+    }
+
+    final JobResultCorrections corrections = new JobResultCorrections();
+    corrections.setAssignee(
+        getStringOrEmpty(request, r -> r.getResult().getCorrections().getAssignee()));
+    corrections.setDueDate(
+        getStringOrEmpty(request, r -> r.getResult().getCorrections().getDueDate()));
+    corrections.setFollowUpDate(
+        getStringOrEmpty(request, r -> r.getResult().getCorrections().getFollowUpDate()));
+    corrections.setCandidateGroups(
+        getStringListOrEmpty(request, r -> r.getResult().getCorrections().getCandidateGroups()));
+    corrections.setCandidateUsers(
+        getStringListOrEmpty(request, r -> r.getResult().getCorrections().getCandidateUsers()));
+    corrections.setPriority(
+        getIntOrDefault(request, r -> r.getResult().getCorrections().getPriority(), -1));
+    return jobResult.setCorrections(corrections);
```

Comment:
👍 I agree with the above.

Reply:
💭 @ce-dmelnych @korthout can it be the case that user explicitly sets the value of the corrected field, for example, `assignee` to be empty and to override the existing `assignee` value?

Alternatively: potentially the list of `correctedAttributes` can be calculated on a java client side. Ex.: when user sets some correction the client also records this attribute as corrected.

Reply:
>can it be the case that user explicitly sets the value of the corrected field, for example, assignee to be empty and to override the existing assignee value?

@ana-vinogradova-camunda Yes. This should be aligned with the User Task update and the Job update. They both allow using `null` to mean unchanged and a value to mean changed to. For strings, an empty string `""` is allowed to intent unassign.

>Alternatively: potentially the list of correctedAttributes can be calculated on a java client side. Ex.: when user sets some correction the client also records this attribute as corrected.

The API doesn't need the `correctedAttributes`. This field only exists because the records in our protocol don't allow `null` values, but the API does. So we can calculate the `correctedAttributes` in the gateway, similar to example shared above by Dima.

Reply:
💭 For such cases, the user can provide a default value (e.g for the `assignee` property -> empty string value `""`) to reset it to the default via corrections. We can highlight this approach in the documentation (Camunda/REST-API). 

Additionally, on the Zeebe Java client side, we could introduce a convenient `reset<PropertyName>()` method (e.g `resetAssignee()`) alongside the ordinary `assignee(final String assignee)` builder method. This method would handle the default (empty string) setting logic internally, making it easier for users to reset fields.

What do you think about this approach, @ana-vinogradova-camunda, @korthout?

Reply:
>resetAssignee()) alongside the ordinary assignee(final String assignee) builder method. This method would handle the default (empty string) setting logic internally, making it easier for users to reset fields.

@ce-dmelnych I worry that `reset` would be interpreted as rollback to before the transition. We currently don't support this conveniently, as the listener doesn't have access to the previous value, nor is any api to specify data rollbacks. It would be an interesting addition, when we add such functionality.